core/controller/src/main/scala/whisk/core/controller/Authenticate.scala

```diff
@@ -69,8 +73,17 @@ trait Authenticate extends Logging {
                     }
                 }
             } onComplete {
-                case Success(s) =>
-                    promise.success(s)
+                case Success(Some(s)) =>
+                    Try {
+                        Identity(s.subject, EntityName(s.subject()), s.authkey, Privilege.ALL)
+                    } match {
+                        case Success(id) => promise.success(Some(id))
+                        case Failure(t) =>
+                            error(this, s"failed to created identity profile: $t")
+                            promise.failure(t)
+                    }
+                case Success(None) =>
+                    promise.success(None)
```

Comment:
I believe we can get rid of the Promise, will try to provide a patch proposal later.